### PR TITLE
add no-proxy check to eslint rules

### PR DESCRIPTION
**Pull request by rianfowler**
## Description

- added new eslint to prohibit to use of Proxy
- added exception to build script plugin 

## Screenshots
Screen Shot 2019-10-07 at 4 10 27 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -140,6 +140,7 @@
     "eslint-config-airbnb": "^15.1.0",
     "eslint-config-prettier": "^3.1.0",
     "eslint-import-resolver-babel-module": "^5.0.1",
+    "eslint-plugin-fp": "^2.3.0",
```
Comment (rianfowler) [inline package.json]: first result with a no-proxy rule. this plugin has some [other rules](https://github.com/jfmengels/eslint-plugin-fp#rules) that might be useful

File: src/site/stages/build/drupal/load-saved-flags.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const fs = require('fs-extra');
  *                   an error if a flag is called without existing
  */
 function useFlags(rawFlags, shouldLog = true) {
+  // eslint-disable-next-line fp/no-proxy
```
Comment (cvalarida) [inline src/site/stages/build/drupal/load-saved-flags.js]: It's worth noting that this isn't browser code, hence the exception. Node doesn't mind proxies.

**Review by cvalarida — APPROVED** [review]
Looks good. :+1:

**Comment by cvalarida** [comment]
Let the record show that this PR is specifically to prohibit the use of proxies because IE doesn't support it. Because we support IE (:grimacing: okay, I mean we support the users...you understand me, I know you do), we can't use proxies.

The build code, on the other hand, is all in Node, so proxies won't be a problem there.

**Pull request merged**